Handle HTTP 422 response to OAuth Device flow detection

mislav opened the pull request:
If HTTP 422 is encountered, assume that OAuth Device Flow is unavailable and fall back to OAuth app authorization flow.

mislav committed:
Handle HTTP 422 response to OAuth Device flow detection

If HTTP 422 is encountered, assume that OAuth Device Flow is unavailable
and fall back to OAuth app authorization flow.

tierninho commented:
So I am getting mixed messages where the browser confirms my login and CLI denies me.

![Screen Shot 2020-08-31 at 11 20 46 AM](https://user-images.githubusercontent.com/14828183/91769961-0aed0900-eb7c-11ea-89aa-1d03cf12f5ff.png)

mislav committed:
Ensure correct ANSI color output during OAuth flow on Windows

We used to write directly to `os.Stderr`, but we first need to convert
that into a colorable stream.

mislav committed:
Validate hostname as entered in the auth prompt

mislav committed:
Validate the `--hostname` flag value

mislav committed:
Only validate `--hostname` when flag was provided

mislav committed:
Tweak `gh config set -h` output format on `auth login`

mislav commented:
I've pushed changes that:
1. Disallow invalid input (e.g. trailing slash) for hostname during `auth login`;
2. Ensures correct color output during auth flow on Windows.

tierninho commented:
> Disallow invalid input (e.g. trailing slash) for hostname during auth login;
>Ensures correct color output during auth flow on Windows.

Both looking good! 🎨
